### ISSUE TITLE
Migrate publishing from Travis/Bintray to GitHub-Actions/Sonatype

### ISSUE THREAD

**Issue by octonato** [akka/akka-meta#149]
The minimal goal for each migration is that a project no longer publishes via Bintray. A project must publish by tag to Sonatype. Ideally, the migration of the CI build includes moving away from whatever CI it currently uses and adopts GitHub Actions.

Current decision is that we will only migrate when there is an immediate need, ie: just before a release. Also, to keep things small and avoid new issues we don't migrate everything to GitHub Actions but only the releasing part (publish to gustav, whitesource and publishing to sonatype).

the fastest way to achieve the minimum (gustav, whitesoiurce, publish to sonatype) is to copy [this workflow](https://github.com/akka/akka-persistence-jdbc/blob/master/.github/workflows/release.yml), adjust it and remove any publish-related job from travis.

- [x] https://github.com/akka/akka-grpc (related to https://github.com/akka/akka-grpc/pull/1325) 
- [x] https://github.com/akka/akka-http (done sonatype)
- [x] https://github.com/akka/akka-paradox 
- [x] https://github.com/playframework/playframework (done sonatype)
- [x] https://github.com/playframework/twirl (use latest interplay and run a release)
- [x] https://github.com/playframework/play-json
- [x] https://github.com/playframework/play-slick
- [x] https://github.com/playframework/play-ws
- [x] https://github.com/playframework/play-file-watch
- [x] https://github.com/playframework/cachecontrol
- [x] https://github.com/playframework/omnidoc
- [x] https://github.com/playframework/interplay (done sonatype)
- [x] https://github.com/lagom/lagom (done sonatype)

The following need migrating away from Bintray and are also affected by #150 (migration to GH Actions required to prevent dockerhub rate-limiting issues in Travis)

- [x] https://github.com/akka/akka-management (in progress https://github.com/akka/akka-management/pull/889)
- [x] https://github.com/akka/akka-persistence-jdbc
- [x] https://github.com/akka/akka-persistence-cassandra
- [x] https://github.com/akka/akka-persistence-spanner
- [ ] https://github.com/akka/akka-persistence-couchbase
- [x] https://github.com/akka/alpakka (sonatype done, missing GH Actions)
- [x] https://github.com/akka/alpakka-kafka 
- [x] https://github.com/akka/akka-projection

**Comment by ignasi35** [comment]
> `playframework/interplay` (we probably can take the opportunity to retire this one)

I don't think we should pay the cost of that retirement now.

**Comment by ihostage** [comment]
- [x] https://github.com/playframework/play-soap (moved to https://github.com/playframework/play-soap/issues/246)

**Comment by octonato** [comment]
@ignasi35, I didn't check them all, but check this https://github.com/playframework/play-ws/blob/master/project/plugins.sbt#L4

Those project do use bintray, either directly or indirectly via interplay. If they are using interplay, I would prefer to step out and just use sbt-ci-release. That's better than having another interplay release just because of that.

**Comment by octonato** [comment]
@ihostage, would you do the necessary for play-soap? 

I can add the necessary keys to Travis.

**Comment by ignasi35** [comment]
> @ignasi35, I didn't check them all, but check this [https://github.com/playframework/play-ws/blob/master/project/plugins.sbt#L4](https://github.com/playframework/play-ws/blob/master/project/plugins.sbt?rgh-link-date=2021-04-19T11%3A39%3A43Z#L4)
>
> Those project do use bintray, either directly or indirectly via interplay. If they are using interplay, I would prefer to step out and just use sbt-ci-release. 

Of course! 🤦🏼‍♂️

There are a handful of projects in the play ecosystem that (1) already moved away from interplay, and (2) used bintray as the staging environment before pushing to sonatype.
 

> That's better than having another interplay release just because of that.

There's no need for another interplay release in any case since latest interplay 2.8.x and 3.0.x already moved away from bintray.

**Comment by ignasi35** [comment]
BTW, I think whatever effort we put in migrating the reports listed in the description of this issue it should include both `master` and stable branches!

**Comment by octonato** [comment]
I'm wondering what could be a quick win strategy here. I think in general we want to move to GH Actions because of the many issues we encounter (and still encounter) with Travis. 

A quick path for the bintray issue could be to only move the release to GH Actions for now. 

I don't want to move everything to GH Action right now and I don't want to start configure Travis with the credentials for sbt-ci-release and a month later redo it for GH Actions. 

In other projects, we the release.yml workflow only works in `master` after a PR merge or after tagging. It will then publish the artifacts (snapshots or release), run whitesource and then publish docs to gustav. 

Bottom line is, we can use GH Actions for the release and Travis for the PR tests.

**Comment by ihostage** [comment]
> @ihostage, would you do the necessary for play-soap?
> I can add the necessary keys to Travis.

@octonato Yes, of course. I created a separate issue for Play SOAP (https://github.com/playframework/play-soap/issues/246) and do that when will find time for that.

**Comment by ignasi35** [comment]
> I'm wondering what could be a quick win strategy here. 

TBQH, I think the best option is to not migrate anything and just migrate if/when required. Something _as simple as_ migrating to GH Actions is already surfacing issues. See the ongoing efforts for [lagom](https://github.com/lagom/lagom/pull/3216) and [akka-grpc](https://github.com/akka/akka-grpc/pull/1325#issuecomment-822393701).

**Comment by octonato** [comment]
@ignasi35, I updated the description with the conclusion of our discussion. 

I will continue with the akka-management anyway because it's suffering from dockerhub rate limiting. All the other projects follow your suggestion to only do when needed.

@ihostage, that doesn't include play-soap. It's up to you when you want to migrate it. Let me know when I need to generate a gpg key for publishing and setup the secrets for you.

**Comment by ignasi35** [comment]
@octonato we should put akka-persistence-cassandra (and probably all other persistence plugins)  on the list to migrate to GH since they use `docker` on the CI.

**Comment by octonato** [comment]
> we should put akka-persistence-cassandra (and probably all other persistence plugins) on the list to migrate to GH since they use docker on the CI.

Ok, I will change the description. First a list of ASAP must have because of Docker. Those a GH Action and Sonatype. 

Then the list of TODOs when we need/time allow.

**Comment by mkurz** [comment]
Please add play-mailer to the list:
- [x] https://github.com/playframework/play-mailer

**Comment by ignasi35** [comment]
I have split this issue in two so #149 stays focused on migrating away from Bintray.

#150, OTOH, focuses on migrating away from Travis, specially those jobs that need `docker` to run.

**Comment by ennru** [comment]
We're good here.
